File: algorithm_practice_python/leetcode/arrays101/SortArrayByParity.py
from typing import List


class SortArrayByParity:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        slow = 0
        for fast in range(len(A)):
            if A[fast] % 2 == 0:
                A[slow], A[fast] = A[fast], A[slow]
                slow += 1
        return A

File: algorithm_practice_java/src/main/java/leetcode/arrays101/SortArrayByParity.java
package leetcode.arrays101;

/**
 * url: https://leetcode.com/explore/learn/card/fun-with-arrays/511/in-place-operations/3260/
 */
public class SortArrayByParity {
    public static int[] sortArrayByParity(int[] A) {
        // even first, odd last and no matter orders
        int slow = 0;
        for (int fast = 0; fast < A.length; fast++) {
            if (A[fast] % 2 == 0) {
                int temp = A[slow];
                A[slow] = A[fast];
                A[fast] = temp;
                slow++;
            }
        }
        return A;
    }
}
